components/family-studies/api/src/main/java/org/phenotips/studies/family/script/FamilyScriptService.java

```diff
@@ -53,94 +53,118 @@
+     * @param patientId id of the patient
+     * @return DocumentReference to the family, or null if doesn't exist
      */
-    public DocumentReference getPatientsFamily(XWikiDocument patient)
+    public DocumentReference getPatientsFamily(String patientId)
     {
-        try {
-            XWikiDocument doc = this.utils.getFamilyDoc(patient);
-            return doc != null ? doc.getDocumentReference() : null;
-        } catch (XWikiException ex) {
-            this.logger.error(FAMILY_NOT_FOUND, ex.getMessage());
+        Patient patient = this.patientRepository.getPatientById(patientId);
+        Family family = this.familyRepository.getFamilyForPatient(patient);
+        if (family == null) {
+            return null;
         }
-        return null;
+        return family.getDocumentReference();
     }
 
     /**
-     * Gets a family with `id` (given a patient id, first finds the family the patient belongs to) and returns info
-     * about the family.
+     * Gets a family id or patient id. If the id is a patient id, finds the patient's family and returns a family
+     * information JSON. If the id is a family id, returns the family information.
      *
-     * @param id must be a valid family id or a patient id (patient must belong to a family)
-     * @return family id and all members that belong to it
+     * @param id family id or the id of a patient who belongs to a family
+     * @return JSON data structure with information about the family
      */
     public JSON getFamilyInfo(String id)
     {
-        try {
-            XWikiDocument doc = this.utils.getFromDataSpace(id);
-            XWikiDocument familyDoc = this.utils.getFamilyDoc(doc);
-            return this.familyInformation.getBasicInfo(familyDoc);
-        } catch (XWikiException ex) {
-            this.logger.error(FAMILY_NOT_FOUND, ex.getMessage());
+        Family family = null;
+
+        Patient patient = this.patientRepository.getPatientById(id);
+        if (patient != null) {
+            // id belonged to a patient. Get patient's family
+            family = this.familyRepository.getFamilyForPatient(patient);
+        } else {
+            // id is not a patient's id. Check if it is a family id
+            family = this.familyRepository.getFamilyById(id);
+        }
+
+        if (family == null) {
+            // id is not a family id, nor a patient's id
+            this.logger.error("getFamilyInfo, id:[{}]. Id does not identify a family or a patient", id);
             return new JSONObject(true);
         }
+
+        return family.getInformationAsJSON();
+    }
+
+    /**
+     * Family page should aggregate medical reports of all its members.
+     *
+     * @param familyId if of family to get reports for
+     * @return patient ids mapped to medical reports, which in turn are maps of report name to its link
+     */
+    public Map<String, Map<String, String>> getReports(String familyId)
```

Comment:
Don't repeat things that can be accessed from a `Family` directly. The script service should only provide access to creating and retrieving `Family` objects, all other things should be done through the family API.